tasks/DroneFlightControlTask.cpp

```diff
@@ -0,0 +1,1456 @@
+}
+
+bool DroneFlightControlTask::takeoffFinishedCheck()
+{
+    while (mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>() ==
+               VehicleStatus::DisplayMode::MODE_ASSISTED_TAKEOFF ||
+           mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>() ==
+               VehicleStatus::DisplayMode::MODE_AUTO_TAKEOFF)
+    {
+        sleep(1);
+    }
+    return ((mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>() ==
+             VehicleStatus::DisplayMode::MODE_P_GPS) ||
+            (mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>() ==
+             VehicleStatus::DisplayMode::MODE_ATTITUDE))
+               ? true
+               : false;
+}
+
+bool DroneFlightControlTask::checkActionStarted(uint8_t mode)
+{
+    int actionNotStarted = 0;
+    int timeoutCycles = 20;
+    while (mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>() != mode &&
+           actionNotStarted < timeoutCycles)
+    {
+        actionNotStarted++;
+        Platform::instance().taskSleepMs(100);
+    }
+    if (actionNotStarted == timeoutCycles)
+    {
+        DERROR("Start actions mode %d failed, current DISPLAYMODE is: %d ...", mode,
+               mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>());
+        return false;
+    }
+    else
+    {
+        DSTATUS("DISPLAYMODE: %d ...", mode);
+        return true;
+    }
+}
+
+bool DroneFlightControlTask::landFinishedCheck(void)
+{
+    while (mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>() ==
+               VehicleStatus::DisplayMode::MODE_AUTO_LANDING &&
+           mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_FLIGHT>() ==
+               VehicleStatus::FlightStatus::IN_AIR)
+    {
+        Platform::instance().taskSleepMs(1000);
+    }
+
+    return ((mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>() !=
+                 VehicleStatus::DisplayMode::MODE_P_GPS ||
+             mSetup.vehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_DISPLAYMODE>() !=
+                 VehicleStatus::DisplayMode::MODE_ATTITUDE))
+               ? true
+               : false;
+}
+
+Telemetry::Vector3f DroneFlightControlTask::vector3FSub(const Telemetry::Vector3f &vectorA,
+                                                        const Telemetry::Vector3f &vectorB)
+{
+    Telemetry::Vector3f result;
+    result.x = vectorA.x - vectorB.x;
+    result.y = vectorA.y - vectorB.y;
+    result.z = vectorA.z - vectorB.z;
+    return result;
+}
+
+void DroneFlightControlTask::horizCommandLimit(float speedFactor, float &commandX,
+                                               float &commandY)
+{
+    if (fabs(commandX) > speedFactor)
+        commandX = signOfData<float>(commandX) * speedFactor;
+    if (fabs(commandY) > speedFactor)
+        commandY = signOfData<float>(commandY) * speedFactor;
+}
```

Comment:
maybe return a pair with commandX and commandY instead of passing them as reference arguments?